config/round_config/config.go

```diff
@@ -0,0 +1,57 @@
+package round_config
+
+import (
+	"time"
+
+	"go.uber.org/zap/zapcore"
+)
+
+const (
+	defaultEpochDuration = 30 * time.Second
+	defaultPhaseShift    = 15 * time.Second
+	defaultCycleGap      = 10 * time.Second
+)
+
+type Config struct {
```

Comment:
I had to move it into a separate package to avoid cyclic dependency: `package config` depends on `package service` which depends on `package round_config`. :(

Reply:
Hm... there seems to be a weird relationship between a few packages:

- both `server` and `rpc` depend on `config`
- `server` depends on `rpc`
- Moving `config` into `server` (imo it belongs there) causes a cyclic dependency

This can be avoided: `rpc` doesn't need to depend on `config`. It only uses 2 fields out of the whole config object and they could be passed explicitly (`PhaseShift` and `CycleGap`).

`config` and `server` both depend on `round_config`, but `registration` as well (which is imported by `server`). This I guess is the reason why you split them into two.

`round_config` could be merged with the other config if the methods defined on `round_config.Config` would be functions defined in the `registration` package (only receiving the relevant fields).

Wdyt?

Reply:
>Moving config into server (imo it belongs there)

Makes sense :+1:

>config and server both depend on round_config, but registration as well (which is imported by server). This I guess is the reason why you split them into two.

Yes, exactly.

>round_config could be merged with the other config if the methods defined on round_config.Config would be functions defined in the registration package (only receiving the relevant fields).

The `round_config.Config` is not part of the `registration` package because it is used by both `registration` and `service` packages. The round start and end calculations are complex enough that I prefer to have them in a single place rather than inline everywhere and risk making a mistake.

Reply:
Done